Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3271,7 +3277,7 @@ void WebPageProxy::processNextQueuedMouseEvent()
 
     LOG_WITH_STREAM(MouseHandling, stream << "UIProcess: sent mouse event " << eventType << " (queue size " << internals().mouseEventQueue.size() << ")");
     m_process->recordUserGestureAuthorizationToken(event.authorizationToken());
-    send(Messages::WebPage::MouseEvent(event, sandboxExtensions));
+    send(Messages::WebPage::MouseEvent(m_mainFrame->frameID(), event, sandboxExtensions));
```

Comment:
If we really want to make mouse handling frame specific, then we should probably be sending the to the WebFrame no? Instead of still sending it to the WebPage with an additional frameID. Note that this will matter even more in the future with process-per-frame.

Reply:
I could do this, but there is still some state related to mouse events on `WebPage` that I don’t think we would want to move to `WebFrame`. Also, I was planning for the life cycle of the event to still be tracked by page, not frame. i.e. `WebPageProxy::didReceiveEvent()` would be called once for each handled event, and there would be a separate function to propagate the mouse event through processes.

Reply:
This is how we send messages to the WebFrame.  It turned out to be too problematic to make the WebFrame a message receiver.